libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -15,6 +15,8 @@
 
 extern AP_HAL::HAL& hal;
 
+Compass Compass::frontend;
+
```

Comment:
@lucasdemarchi as I said before, both ways are correct. There is another place where singleton is implemented: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_HAL_PX4/HAL_PX4_Class.cpp#L315 the only thing we need to discuss is which one should be standard. Personally I don't have a preference, I fine with any. @tridge can weigh in what he thinks and I will make the change if need be.